client/src/modules/shopping-list/model/actions.js

```diff
@@ -93,3 +117,31 @@ export const fetchShoppingListsMetaData = () => dispatch => {
       );
     });
 };
+
+export const deleteList = (id, successRedirectCallback) => dispatch => {
+  dispatch(deleteListRequest());
+  deleteData(`${ENDPOINT_URL}/shopping-lists/${id}/delete`)
+    .then(resp =>
+      resp.text().then(message => {
+        if (resp.ok) {
+          dispatch(deleteListSuccess(id));
+          createNotificationWithTimeout(
+            dispatch,
+            NotificationType.SUCCESS,
+            message
+          );
+          successRedirectCallback();
+        } else {
+          const error = new Error(message);
```

Comment:
try to put to deleteData:
  if (response.status >= 400 && response.status < 600) {
    throw new Error('Bad response from server');
  }
to handle errors in catch clause